crates/router/src/core/payments/operations/payment_update.rs

```diff
@@ -719,6 +711,26 @@ impl<F: Clone> UpdateTracker<F, PaymentData<F>, api::PaymentsRequest> for Paymen
             .payment_intent
             .statement_descriptor_suffix
             .clone();
+
+        let billing_details = payment_data
+            .address
+            .get_payment_billing()
+            .async_and_then(|_| async {
+                create_encrypted_data(
+                    key_store,
+                    payment_data.address.get_payment_billing().cloned(),
+                )
```

Comment:
create another function which will take the data and encrypt it, and rename this function to indicate that this encrypts optional data

Reply:
Sure this will be taken up in the very next PR.